webapp/lib/app/users/controller.dart

```diff
@@ -0,0 +1,242 @@
+                editedConfig[c.docId].remove(permissionKey);
+              }
+              if (editedConfig[c.docId].isEmpty) {
+                editedConfig.remove(c.docId);
+              }
+            }
+            
+            var valueToUpdate = modifiedObj[permissionKey];
+            var valueType = valueToUpdate.runtimeType.toString();
+            if (valueType == 'bool') {
+              _view.updateBoolPermission(c.docId, permissionKey, valueToUpdate);
+            } else if (valueType == 'String') {
+              _view.updateStringPermission(c.docId, permissionKey, valueToUpdate);
+            } else if (valueType == 'List<String>') {
+              _view.updateListStringPermission(c.docId, permissionKey, valueToUpdate);
+            } else {
+              log.error("Unknown data type for ${c.docId}, ${permissionKey}, ${valueToUpdate}");
+            }
+          });
+        });
+      }
+
+      // todo: handle removal of user
+      _view.enableSaveButton(editedConfig.isNotEmpty);
+    });
+  }
+
+  void command(action, [Data data]) {
+    if (action is! UsersAction) {
+      super.command(action, data);
+      return;
+    }
+
+    switch (action) {
+      case UsersAction.savePermissions:
+        _view.setSaveText("Updating…");
+        _view.enableSaveButton(false);
+        platform.updateUserConfiguration(editedConfig).then((_) {
+          editedConfig.keys.forEach((email) {
+            _view.markAsUnsaved(email, true);
+          });
+          editedConfig = {};
+          _view.setSaveText("Updated!");
+          new Timer(new Duration(seconds: 2), () => _view.setSaveText("Update permissions"));
+        });
+        break;
+
+      case UsersAction.updateBoolPermission:
+        var updateData = data as UpdateBoolPermission;
+        if (isDefaultPermission(updateData.email)) {
+          var userDataObj = defaultConfig.toData();
+          userDataObj[updateData.permissionKey] = updateData.value;
+          defaultConfig = model.UserConfiguration.fromData(userDataObj);
+          for (var email in userConfigs.keys) {
+            if (email != 'default') {
+              var valueToUpdate = userConfigs[email].toData()[updateData.permissionKey] ?? updateData.value;
+              _view.updateBoolPermission(email, updateData.permissionKey, valueToUpdate);
+            }
+          }
+        } else {
+          var userDataObj = userConfigs[updateData.email].toData();
+          userDataObj[updateData.permissionKey] = updateData.value;
+          userConfigs[updateData.email] = model.UserConfiguration.fromData(userDataObj);
+        }
+        editedConfig[updateData.email] = editedConfig[updateData.email] ?? {};
+        editedConfig[updateData.email][updateData.permissionKey] = updateData.value;
+        _view.markAsUnsaved(updateData.email, false);
+        _view.enableSaveButton(editedConfig.isNotEmpty);
+        break;
+
+      case UsersAction.updateStringPermission:
```

Comment:
The code for `update<Type>Permission` is almost identical. I'd be inclined to suggest we have just one user action (`updatePermission`) to which we pass the type of permission - what do you think? (and a similar approach for the view - one entry point to update a permission, but potentially there we keep private methods for each type)